### PR TITLE
implement ABCInterval

### PR DESCRIPTION
**Pull request by jbrockmendel**
Avoid python-space import in lib

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/dtypes/generic.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def _check(cls, inst):
 ABCPeriod = create_pandas_abc_type("ABCPeriod", "_typ", ("period", ))
 ABCDateOffset = create_pandas_abc_type("ABCDateOffset", "_typ",
                                        ("dateoffset",))
+ABCInterval = create_pandas_abc_type("ABCPeriod", "_typ", ("interval", ))
```
Comment (jschendel) [inline pandas/core/dtypes/generic.py]: The first argument should be "ABCInterval" instead of "ABCPeriod", right? Or am I missing something?

Comment (gfyoung) [reply pandas/core/dtypes/generic.py]: @jschendel : I don't believe you are.

Comment (gfyoung) [reply pandas/core/dtypes/generic.py]: @jbrockmendel : Do we need any tests for this new `ABCType` ?

**Review by jreback — CHANGES_REQUESTED** [review]
as per comments

File: pandas/core/dtypes/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/generic.py]: Yah typo, will fix, good catch.

**Review by gfyoung — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**